src/main/java/io/leonis/algieba/spatial/HydrodynamicPotentialField.java

```diff
@@ -0,0 +1,130 @@
+package io.leonis.algieba.spatial;
+
+import io.leonis.algieba.calculus.LocalLinearLineIntegral;
+import io.leonis.algieba.geometry.Rotation;
+import java.util.function.UnaryOperator;
+import lombok.Value;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+import org.nd4j.linalg.ops.transforms.Transforms;
+
+/**
+ * The Class HydrodynamicPotentialField.
+ *
+ * This class represents a hydrodynamic {@link PotentialField} as proposed in section 2.1 of
+ * <a href="https://www.ri.cmu.edu/pub_files/pub3/kim_jin_oh_1992_2/kim_jin_oh_1992_2.pdf">this paper</a>.
+ *
+ * @author Rimon Oz
+ */
+@Value
+public class HydrodynamicPotentialField
+    implements LocalLinearLineIntegral, Rotation, PotentialField {
+  /**
+   * A vector pointing to the origin.
+   */
+  private final INDArray origin;
+  /**
+   * Height in potential.
+   */
+  private final double height;
+  /**
+   * Width of the field.
+   */
+  private final double width;
+  /**
+   * Length of the field.
+   */
+  private final double length;
+  /**
+   * Angle in radians.
+   */
+  private final double angle;
+
+  /**
+   * See <a href="http://www.wolframalpha.com/input/?i=ln(x%5E2+%2B+y%5E2)">this equation.</a>
+   *
+   * @param positionVector The position vector at which to compute the potential.
+   */
+  @Override
+  public double getPotential(final INDArray positionVector) {
+    return this.height
+        * Math.log(Transforms.pow(this.toLocalFrame(positionVector), 2).sumNumber().doubleValue())
+        / (2 * Math.PI);
+  }
+
+  /**
+   * See <a href="http://www.wolframalpha.com/input/?i=derivate+of+ln(x%5E2+%2B+y%5E2)">this equation.</a>
+   *
+   * @param positionVector The position vector at which to compute the force.
+   */
+  @Override
+  public INDArray getForce(final INDArray positionVector) {
+    final INDArray localPositionVector = this.toLocalFrame(positionVector);
+
+    return this.toGlobalFrame(localPositionVector.mul(2d)
+        .div(Transforms.pow(localPositionVector, 2).sumNumber()));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public INDArray toLocalFrame(final INDArray positionVector) {
+    return this.planarCartesian(positionVector.sub(this.getOrigin()), this.getAngle())
+        .div(Nd4j.create(new double[]{this.getLength(), this.getWidth()}, new int[]{2, 1}));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public INDArray toGlobalFrame(final INDArray positionVector) {
+    return this.planarCartesian(
+        positionVector
+            .mul(Nd4j.create(new double[]{this.getLength(), this.getWidth()}, new int[]{2, 1})),
+        -1 * this.getAngle());
```

Comment:
This reminds me; i think the angle is ambiguous (at least for me as ultra-leek) and the description `The angle in radians` didn't help a lot for me. It's unrelated to this PR though.

Reply:
i could be a bit clearer and will expand the javadoc